dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EnrollmentExistingEnrollmentsValidationHook.java

```diff
@@ -0,0 +1,249 @@
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramInstance;
+import org.hisp.dhis.program.ProgramInstanceQueryParams;
+import org.hisp.dhis.trackedentity.TrackedEntityInstance;
+import org.hisp.dhis.trackedentitycomment.TrackedEntityComment;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Coordinate;
+import org.hisp.dhis.tracker.domain.Enrollment;
+import org.hisp.dhis.tracker.domain.EnrollmentStatus;
+import org.hisp.dhis.tracker.domain.Note;
+import org.hisp.dhis.tracker.preheat.PreheatHelper;
+import org.hisp.dhis.tracker.report.TrackerErrorCode;
+import org.hisp.dhis.tracker.report.TrackerErrorReport;
+import org.hisp.dhis.tracker.report.ValidationErrorReporter;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.util.DateUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.hisp.dhis.tracker.report.ValidationErrorReporter.newReport;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Component
+public class EnrollmentExistingEnrollmentsValidationHook
+    extends AbstractTrackerValidationHook
+{
+
+    @Override
+    public int getOrder()
+    {
+        return 103;
+    }
+
+    @Override
+    public List<TrackerErrorReport> validate( TrackerBundle bundle )
+    {
+        if ( bundle.getImportStrategy().isDelete() )
+        {
+            return Collections.emptyList();
+        }
+
+        ValidationErrorReporter reporter = new ValidationErrorReporter( bundle, this.getClass() );
+
+        User actingUser = bundle.getPreheat().getUser();
+
+        for ( Enrollment enrollment : bundle.getEnrollments() )
+        {
+            reporter.increment( enrollment );
+
+            if ( EnrollmentStatus.CANCELLED == enrollment.getStatus() )
+            {
+                continue;
+            }
+
+            Program program = PreheatHelper.getProgram( bundle, enrollment.getProgram() );
+            OrganisationUnit organisationUnit = PreheatHelper.getOrganisationUnit( bundle, enrollment.getOrgUnit() );
+            TrackedEntityInstance trackedEntityInstance = PreheatHelper
+                .getTrackedEntityInstance( bundle, enrollment.getTrackedEntityInstance() );
+
+            // NOTE: maybe this should qualify as a hard break, on the prev hook (required properties).
+            if ( program == null || organisationUnit == null || trackedEntityInstance == null )
+            {
+                continue;
+            }
+
+            if ( EnrollmentStatus.COMPLETED == enrollment.getStatus()
+                && Boolean.FALSE.equals( program.getOnlyEnrollOnce() ) )
+            {
+                continue;
+            }
+
+            // Sort out only the programs the importing user has access too...
+            // Stian, Morten H.  NOTE: How will this affect validation? If there is a conflict here but importing user is not allowed to know,
```

Comment:
We can't allow data to get in if its supposed to be blocked. So in this case we should fail the validation. The question is what kind of error we return. We need to make sure we dont imply other data. However, I think in this case, it doesnt matter (See comment below, wait for Abyot's confirmation)